docs/providers/aws/guide/plugins.md

```diff
@@ -195,6 +195,41 @@ class Deploy {
 module.exports = Deploy;
 ```
 
+### Custom variable getters
+
+As of version 1.52.0 of Serverless framework, you can officially implement your own variable types
+for use in `serverless.yml` config files.
+
+Example:
+
+```javascript
+'use strict';
+
+class EchoTestVarPlugin {
+  constructor() {
+    getEchoTestValue(src) {
+      return src.slice(5);
+    }
+    getDependentEchoTestValue(src) {
+      return src.slice(5);
+    }
+    this.variableResolvers = {
+      echo: this.getEchoTestValue,
+      // if a variable type depends on profile/stage/region/credentials, to avoid infinite loops in
+      // trying to resolve variables that depend on themselves, specify as such by setting a
+      // dependendServiceName property on the variable getter
+      echoStageDependent: : {
+        resolver: this.getDependentEchoTestValue,
+        serviceName: 'echo that isnt prepopulated',
```

Comment:
I was same way confused by it, when I looked first into PR (and internals of var).

This is about name of a service through which we resolve that variables (e.g. in ssm vars it can be 'SSM').

After acknowledging that, I was not that confused anymore. However we're definitely open for any better name proposals, e.g. we can narrow specificity of it, and make it `resolverServiceName` (?)

Reply:
Oh!  Yes, I totally missed that, thank you.

One might characterize it as something like "the service that cannot be used before prepopulation".  This new understanding brings up a question of whether the variable dependency (to reference my comment about that) is a service but I don't have a counter example on top of mind.

Perhaps an alternative would be something related to the dependency nature or the variable's type.  Maybe, as per your next comment, this too could be removed?